pkg/gofr/datasource/pubsub/mqtt/mqtt_test.go

```diff
@@ -561,3 +562,18 @@ func getMockMQTT(t *testing.T, conf *Config) (*gomock.Controller, *MQTT, *MockCl
 
 	return ctrl, mq, mockClient, mockMetrics, mockToken
 }
+
+func TestMQTT_Close(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	ctrl, client, mockMQTT, _, _ := getMockMQTT(t, mockConfigs)
+	defer ctrl.Finish()
```

Comment:
This comment is out of the PR scope

But, I'm seeing a lot of defer like this one

Why not, adding a t.Cleanup to getMockMQTT and gomock.NewController

And remove all this.

I'm just starting the discussion about it,to see if there could be an agreement on doing so.

I would have to be done in another PR, and we should consider creating an issue referencing this thread

Reply:
Nice thought we can do it in a separate issue

Reply:
I opened

- https://github.com/gofr-dev/gofr/issues/855